src/watchdog.rs

```diff
@@ -0,0 +1,185 @@
+#[cfg(esp_idf_esp_task_wdt)]
+pub mod task {
+    //! ## Example
+    //!
+    //! ```rust, ignore
+    //! # fn main() -> Result<()> {
+    //! let mut watchdog = TaskWatchdog::take()?.unwrap();
+    //!
+    //! watchdog.reconfigure(WatchdogConfig {
+    //!     duration: Duration::from_secs(2),
+    //!     panic_on_trigger: true,
+    //!     ..Default::default()
+    //! })?;
+    //!
+    //! let task = Task::current();
+    //!
+    //! task.set_watchdog(&mut watchdog)?;
+    //!
+    //! loop {
+    //!     task.reset_watchdog()?;
+    //!     unsafe { vTaskDelay(1) };
+    //! }
+    //! # }
+    //! ```
+
+    use esp_idf_sys::*;
+
+    use crate::cpu::Core;
+
+    pub struct WatchdogConfig {
+        pub duration: core::time::Duration,
+        pub panic_on_trigger: bool,
+        // unused by now
+        pub cores: heapless::Vec<Core, 2>,
+    }
+
+    impl Default for WatchdogConfig {
+        fn default() -> Self {
+            let mut cores = heapless::Vec::new();
+            if CONFIG_ESP_TASK_WDT_CHECK_IDLE_TASK_CPU0 == 1 {
```

Comment:
I'm not sure what idle tasks have to do with this implementation. This is for *our* tasks, right? Not for the two idle tasks of ESP IDF to which we don't have access anyway?

Reply:
Belongs to esp idle tasks. Configures whether the idle tasks should be watched or not. (only configurable in idf version 5.0)